### DISCUSSION TITLE
Dispatch OnBeforeStore event on Store method instead of SaveChanges method?

### DISCUSSION THREAD

**Discussion by montyclt** [ravendb/ravendb#13888] [Ideas]
Hello, the [documentation of the OnBeforeStore method says that it is part of SaveChanges](https://ravendb.net/docs/article-page/5.3/csharp/client-api/session/how-to/subscribe-to-events#onbeforestore), but its names suggest that is calling on Store method, causing a confusion.

Independently of the name confusion, I'm aware that changing it is a breaking change, but I'm in a situation on I need an event that is called on Store method and not on SaveChanges, so I suggest adding a new event called `OnBeforeStoreMethod` or similar.

```cs
session.Advanced.OnBeforeStoreMethod += (_, args) =>
{
    if (args.Entity is Person p)
    {
        // Make some modifications on the Entity, e.g.:
        p.Id = new PersonId(args.DocumentId);
    }
};

var person = new Person("María", "Perez");

await session.StoreAsync(person); // I want to see my entity modified here.
await session.SaveChangesAsync();
```

**Comment by ayende** [comment]
Answered here: https://github.com/ravendb/ravendb/discussions/13889
